ui/src/reusable_ui/components/page_layout/test/PageHeader.test.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import {shallow} from 'enzyme'
+
+import {Page} from 'src/reusable_ui'
+
+describe('PageHeader', () => {
+  let wrapper
+
+  const wrapperSetup = (
+    overrideProps = {},
+    overrideLeft = (
+      <Page.Header.Left>
+        <Page.Title title="Yeehaw" />
+      </Page.Header.Left>
+    ),
+    overrideCenter = (
+      <Page.Header.Center>
+        <p>Center!</p>
+      </Page.Header.Center>
+    ),
+    overrideRight = <Page.Header.Right />
+  ) => {
+    const props = {
+      fullWidth: false,
+      inPresentationMode: false,
+      ...overrideProps,
+    }
+
+    return shallow(
+      <Page.Header {...props}>
+        {overrideLeft}
+        {overrideCenter}
+        {overrideRight}
+      </Page.Header>
+    )
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    wrapper = wrapperSetup()
+  })
+
+  it('mounts without exploding', () => {
+    expect(wrapper).toHaveLength(1)
+  })
+
+  it('should contain matching error text if incorrect child types are passed in', done => {
+    try {
+      wrapper = wrapperSetup({}, <p />, <p />, <p />)
+    } catch (error) {
+      expect(error.toString()).toEqual(
+        'Error: <Page.Header> expected children of type <Page.Header.Left>, <Page.Header.Center>, or <Page.Header.Right>'
+      )
+      done()
+    }
+  })
+
+  it('should contain matching error text if too many Header.Rights exist', done => {
+    try {
+      wrapper = wrapperSetup(
+        {},
+        <Page.Header.Right />,
+        <Page.Header.Right />,
+        <Page.Header.Right />
+      )
+    } catch (error) {
+      expect(error.toString()).toEqual(
```

Comment:
Can also use the [`toThrow`](https://jestjs.io/docs/en/expect#tothrowerror) matcher in Jest